mantidproject/mantid
# Iterator for `SpectrumInfo`

**Description of work.**
Implemented an iterator for `SpectrumInfo`. The iterator can be used on both the C++ side and the Python side. 

This iterator is similar to how the one for `DetectorInfo` works. Once #23226 is merged into master and this PR is as well there is a plan to reduce code duplication by templating the iterator.

**Report to:** @SimonHeybrock 

**To test:**
Not much to test - just make sure the Python tests all pass.


To run the Python tests please run the following command:
`ctest -C Debug -R SpectrumInfoTest -V`

Fixes #23253. 




---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
